piquasso/api/result.py

```diff
@@ -13,36 +13,111 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import List, Tuple, Optional, Union
+import random
 
-from piquasso.api.state import State
+from typing import List, Tuple, Union, Optional
 
 import numpy as np
 
+from .branch import Branch
+from .config import Config
+from .state import State
+from .exceptions import NotImplementedCalculation
 
-class Result:
-    """Class for collecting results."""
 
-    def __init__(
-        self,
-        state: Optional[State] = None,
-        samples: Optional[Union[List[Tuple], "np.ndarray"]] = None,
-    ) -> None:
+class Result:
+    r"""Collects the results of a quantum computation.
+
+    The :class:`~piquasso.api.result.Result` class contains the results of a quantum
```

Comment:
Minor: why reference the class if this is already the docstring of the said class?